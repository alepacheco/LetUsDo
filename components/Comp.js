function Home() {
  return <div>Imported</div>;
}

export default Home;