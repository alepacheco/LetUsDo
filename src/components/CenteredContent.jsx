import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
