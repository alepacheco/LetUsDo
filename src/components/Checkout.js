import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}