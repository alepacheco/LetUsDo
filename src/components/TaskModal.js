import React from 'react';
import Card from 'react-bootstrap/Card';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import CheckoutPopUp from './CheckoutPopUp';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import '../styles/components/TaskModal.scss';
import * as actions from '../actions/taskModalActions';

class TaskModal extends React.Component {
    render() {
        return (
            <div className="center-wrapper">
                <CheckoutPopUp
                    show={this.props.checkoutPopup}
                    onHide={() => this.props.actions.setDialog(false)}
                ></CheckoutPopUp>

                <Card className="blob center-inner">
                    <Card.Body>
                        <Card.Title className="wedo-title left-text">We Do</Card.Title>
                        <Form className="form-wrapper">
                            <Form.Group>
                                <Form.Control
                                    onChange={(e) => this.props.actions.setTaskTest(e.target.value)}
                                    value={this.props.taskText}
                                    className="task-form-input"
                                    as="textarea"
                                    rows="3"
                                    placeholder="What task do you need done?"
                                />
                            </Form.Group>
                        </Form>
                        <div className="get-it-done-button-wrapper">
                            <Button className="get-it-done-button" onClick={() => this.props.actions.setDialog(true)}>Get it done</Button>
                        </div>
                    </Card.Body>
                </Card>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        checkoutPopup: state.taskModal.checkoutPopupOpen,
        taskText: state.taskModal.taskText
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TaskModal);