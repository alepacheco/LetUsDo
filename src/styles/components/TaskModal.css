body {
  background-color: #003f91;
}

textarea,
select,
input,
button,
button:focus {
  outline: none;
  resize: none;
  box-shadow: none !important;
}

.form-wrapper {
  background: none;
  border: none;
}

.center-wrapper {
  text-align: center;

}

.center-inner {
  padding: 16px;
  display: inline-block;
}

.left-text {
  text-align:left;
}

.blob {
  margin-top: 36px;
  width: 50%;

  @media screen and (max-width: 982px) {
      width: 85%;
  }
}

.wedo-title {
  font-size: 38pt;
  font-family: JelleeRoman, Sans-Serif;
  font-weight: normal;
  font-style: normal;
  margin-top: -14px;
  margin-left: -6px;

  @media screen and (max-width: 982px) {
    font-size: 148px;
}
}

.task-form-input {
  @media screen and (max-width: 982px) {
    font-size: 52px;
}
}

.get-it-done-button {
  margin-left: auto;
  @media screen and (max-width: 982px) {
    font-size: 52px;
    border-radius: 14px;

  }
}
.get-it-done-button-wrapper {
  margin: 26px 0 0 0;
  text-align: right;
}
