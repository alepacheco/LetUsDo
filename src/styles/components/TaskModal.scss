.task-modal {
  textarea,
  select,
  input,
  button,
  button:focus {
    outline: none;
    resize: none;
    box-shadow: none !important;
  }

  .form-wrapper {
    background: none;
    border: none;
  }

  .center-inner {
    padding: 16px;
    width: 100%;
  }

  .left-text {
    text-align: left;
  }

  .LetUsDo-title {
    font-size: 38pt;
    font-family: JelleeRoman, Sans-Serif;
    font-weight: normal;
    font-style: normal;
    margin-top: -14px;
    margin-left: -6px;
  }

  .get-it-done-button {
    margin-left: auto;
  }
  .get-it-done-button-wrapper {
    margin: 26px 0 0 0;
    text-align: right;
  }
}
