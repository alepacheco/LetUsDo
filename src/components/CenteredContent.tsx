import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
