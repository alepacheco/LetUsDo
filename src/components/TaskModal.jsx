import React from 'react';
import Card from 'react-bootstrap/Card';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import CheckoutPopUp from './CheckoutPopUp';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import '../styles/components/TaskModal.scss';
import * as actions from '../actions/taskModalActions';
import CenteredContent from './CenteredContent';

export class TaskModal extends React.Component {
  constructor(props) {
    super(props);
    this.onClickGetItDone = this.onClickGetItDone.bind(this);
    this.onChangeTaskDescription = this.onChangeTaskDescription.bind(this);
    this.onHideCheckoutPopUp = this.onHideCheckoutPopUp.bind(this);
  }

  async onClickGetItDone() {
    this.props.actions.setDialog(true);
    // TODO add tracking
  }

  onChangeTaskDescription(event) {
    this.props.actions.setTaskTest(event.target.value);
  }

  onHideCheckoutPopUp() {
    this.props.actions.setDialog(false);
  }

  render() {
    return (
      <div className="task-modal">
        <CenteredContent>
          <CheckoutPopUp show={this.props.checkoutPopup} onHide={this.onHideCheckoutPopUp} />
          <Card className="center-inner">
            <Card.Body>
              <Card.Title className="LetUsDo-title left-text">Let Us Do</Card.Title>
              <Form className="form-wrapper">
                <Form.Group>
                  <Form.Control
                    onChange={this.onChangeTaskDescription}
                    value={this.props.taskText}
                    className="task-form-input"
                    as="textarea"
                    rows="3"
                    placeholder="What task do you need done?"
                  />
                </Form.Group>
              </Form>
              <div className="get-it-done-button-wrapper">
                <Button
                  className="get-it-done-button"
                  onClick={this.onClickGetItDone}
                  disabled={this.props.taskText === ''}
                >
                  Get it done
                </Button>
              </div>
            </Card.Body>
          </Card>
        </CenteredContent>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    checkoutPopup: state.taskModal.checkoutPopupOpen,
    taskText: state.taskModal.taskText
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskModal);

TaskModal.propTypes = {
  taskText: PropTypes.string,
  actions: PropTypes.shape({
    setDialog: PropTypes.func,
    setTaskTest: PropTypes.func
  }),
  checkoutPopup: PropTypes.bool
};
