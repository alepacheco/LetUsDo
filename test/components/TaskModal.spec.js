import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {TaskModal} from '../../src/components/TaskModal';

describe('<TaskModal />', () => {
  it('renders', () => {
    const wrapper = shallow(<TaskModal
      taskText="Sample task"
      actions={{
          setDialog: () => {},
          setTaskTest: () => {}
      }}
      checkoutPopup={false}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
