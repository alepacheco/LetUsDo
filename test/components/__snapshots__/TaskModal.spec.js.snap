// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskModal /> renders 1`] = `
<div
  className="center-wrapper"
>
  <CheckoutPopUp
    onHide={[Function]}
    show={false}
  />
  <Card
    body={false}
    className="blob center-inner"
  >
    <CardBody>
      <CardTitle
        className="wedo-title left-text"
      >
        We Do
      </CardTitle>
      <Form
        className="form-wrapper"
        inline={false}
      >
        <FormGroup>
          <FormControl
            as="textarea"
            className="task-form-input"
            onChange={[Function]}
            placeholder="What task do you need done?"
            rows="3"
            value="Sample task"
          />
        </FormGroup>
      </Form>
      <div
        className="get-it-done-button-wrapper"
      >
        <Button
          active={false}
          className="get-it-done-button"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Get it done
        </Button>
      </div>
    </CardBody>
  </Card>
</div>
`;
